cms/djangoapps/contentstore/views/tests/test_organizations.py

```diff
@@ -32,3 +35,57 @@ def test_organization_list(self):
         self.assertEqual(response.status_code, 200)
         org_names = json.loads(response.content)
         self.assertEqual(org_names, self.org_short_names)
+
+
+@patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+@override_switch(settings.ENABLE_EDLY_ORGANIZATIONS_SWITCH, active=True)
+class TestEdlyOrganizationListing(TestCase):
+    """
+    Verify Organization listing behavior.
+    """
+    @patch.dict('django.conf.settings.FEATURES', {'ORGANIZATIONS_APP': True})
+    def setUp(self):
+        super(TestEdlyOrganizationListing, self).setUp()
+        self.staff = UserFactory(is_staff=True)
+        self.client.login(username=self.staff.username, password='test')
+        self.org_names_listing_url = reverse('organizations')
+
+    def test_without_authentication(self):
+        """
+        Verify authentication is required when accessing the endpoint.
+        """
+        self.client.logout()
+        response = self.client.get(self.org_names_listing_url)
+        self.assertEqual(response.status_code, 302)
+
+    def test_organization_list(self):
+        """
+        Verify that the organization names list API only returns Edly's enabled organizations.
+        """
+
+        studio_site = SiteFactory()
+        edly_organization = EdlyOrganizationFactory(name='Test Edly Organization Name')
+        edly_sub_organization = EdlySubOrganizationFactory(
+            name='Test Edly Sub Organization Name',
```

Comment:
you don't need to provide these unless you need some custom hardcoded value. The factory automatically generates name and slug